### PR TITLE
feat: allow customising or disabling git-push used by manual release

### LINKED ISSUE #1267: Make publish:git configurable?
My project is using `AwsCdkConstructLibrary` with `releaseTrigger: ReleaseTrigger.manual()` set. When running `yarn build` Projen automatically creates a commit, tags it and pushes it to my remote. I'd like more control over this. 

I'm wanting to run a pipeline in Gitlab which runs `yarn release` whenever a commit is pushed to my **main** branch. As `yarn release` pushes automatically I need to be able to tell Gitlab to NOT start another pipeline when this push occurs. Gitlab supports this with the option `-o ci.skip`

Currently publish:git does this:

```
git push --follow-tags origin main
```

I'd like it to do this:

```
git push --follow-tags -o ci.skip origin main
```

A way to override the git command would probably suffice?

### PR DESCRIPTION
**Pull request by willdady**
This adds the ability for a user to customise (or disable) the git-push command which is run as part of a manual release.

closes #1267 

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/release/publisher.ts [hunk 1]
```diff
@@ -753,4 +757,11 @@ export interface GitPublishOptions extends VersionArtifactOptions {
    * @default "main"
    */
   readonly gitBranch?: string;
+  
+  /**
+   * Override git-push command. 
+   * 
+   * Set to null to disable pushing.
+   */
+  readonly gitPushCommand?: string | null;
```
Comment (Chriscbr) [inline src/release/publisher.ts:766]:
```suggestion
  readonly gitPushCommand?: string;
```

Projen is compiled with JSII to other languages where `undefined` and `null` will be treated as equivalent, so distinguishing between them may lead to errors.

How about we instead add a line in the docstring that says e.g. `To disable pushing, use a no-op command such as "true"`?

**Review by Chriscbr — CHANGES_REQUESTED** [review]
Thanks for submitting this PR! Just one requested API change below.

File: src/release/publisher.ts (hunk 1 above)
Comment (willdady) [reply src/release/publisher.ts:766]: Thanks @Chriscbr. How about, `To disable pushing, set to an empty string`? 

I feel `gitPushCommand: true` reads a little weirdly for something which is disabling something.

Comment (Chriscbr) [reply src/release/publisher.ts]: Fair point. I suppose the alternative here is to add another field. This solution sounds good to me. 🙂

**Review by Chriscbr — APPROVED** [review]

**Comment by Chriscbr** [comment]
From the build error it looks like PR code just needs to be linted and the docs updated (`yarn eslint` and `yarn docgen` probably)

**Comment by willdady** [comment]
Build is failing as it's trying to do a git-push to _my_ fork?

**Comment by Chriscbr** [comment]
The actual error is in the build step right before (this is a bit confusing). It should be possible to fix this by merging the latest changes from `main` and re-building

**Comment by willdady** [comment]
> The actual error is in the build step right before (this is a bit confusing). It should be possible to fix this by merging the latest changes from `main` and re-building

@Chriscbr Thanks. I've just done this now.

**Comment by Chriscbr** [comment]
Sorry for the confusion - when I said it will be fixed by "re-building" I meant running a build on your local machine, and then committing any of the changes that it makes (it may update test snapshots or docs) to your branch. If you look at the diff, you can see it incorrectly includes changes to "cdk8sPlusVersion" in API.md that are unrelated to your PR.

**Pull request merged**